### PR TITLE
Supporting reversed ranges in 3d scenes (final)

### PR DESCRIPTION
**Pull request by archmoj**
This PR is the continuation of [PR 3017](https://github.com/plotly/plotly.js/pull/3071).
To minimize merge conflicts with master, the latest changes are fetched and already applied in this new branch.
This PR would allow users to display 3D data in plotly.js scenes that may have one or multiple reversed ranges and resolves [issue 1940](https://github.com/plotly/plotly.js/issues/1940).
The options such as autorange:'reversed' and/or range:[larger, smaller] could be applied in this regard. Most of the fixes to expand this capability are performed at vertex and fragment shaders. And 3 new image tests are added.
@etpinard @alexcjohnson

### CODE REVIEW

**Comment by etpinard** [comment]
> This PR is the continuation of [PR 3017](https://github.com/plotly/plotly.js/pull/3071).



> To minimize merge conflicts with master, the latest changes are fetched and already applied in this new branch.

Great. Thanks for opening a new PR. This makes my life much easier.

Can we close https://github.com/plotly/plotly.js/pull/3071 ?

**Comment by archmoj** [comment]
Yes we could close the other PR. Do you want me to close that?

File: package.json [hunk 1]
```diff
@@ -70,21 +70,21 @@
...
+    "gl-cone3d": "^1.2.0",
     "gl-contour2d": "^1.1.4",
-    "gl-error3d": "^1.0.7",
+    "gl-error3d": "^1.0.8",
     "gl-heatmap2d": "^1.0.4",
-    "gl-line3d": "^1.1.2",
+    "gl-line3d": "^1.1.3",
     "gl-mat4": "^1.2.0",
-    "gl-mesh3d": "^2.0.0",
+    "gl-mesh3d": "2.0.1",
```
Comment (etpinard) [inline package.json]: Let's bring back that `^`, this should be `^2.0.1`.

**Comment by etpinard** [comment]
@archmoj amazing work!

I closed https://github.com/plotly/plotly.js/pull/3071 and made one tiny tiny comment. Once addressed, we'll go ahead and merge this thing.

**Comment by archmoj** [comment]
Thank you @etpinard & @alexcjohnson for the encouragement and help on this very first PR. 
Your idea of using glslify was really helped simplify the code.
I updated the package.json & it looks like there is no additional change appeared in the lock when I used `npm i`. So this should be ready to merge, I hope.

**Pull request merged**